util/SerializeMultiplayerCommon.cpp

```diff
@@ -53,22 +53,72 @@ void SaveGameUIData::serialize(Archive& ar, const unsigned int version)
         & BOOST_SERIALIZATION_NVP(map_zoom_steps_in)
         & BOOST_SERIALIZATION_NVP(fleets_exploring);
 
-    if (version >= 1) {
+    if (version >= 2) {
+        ar & BOOST_SERIALIZATION_NVP(obsolete_ui_event_count);
         ar & BOOST_SERIALIZATION_NVP(ordered_ship_design_ids_and_obsolete);
         ar & BOOST_SERIALIZATION_NVP(ordered_ship_hull_and_obsolete);
         ar & BOOST_SERIALIZATION_NVP(obsolete_ship_parts);
+    }
+
+    // Only workarounds for old versions follow
+    if (version >= 2)
+        return;
+
+    if (!Archive::is_loading::value)
+        return;
+
+    if (version == 1) {
+        obsolete_ui_event_count = 1;
```

Comment:
this is pretty long to keep in the serialize function... maybe move to a separate function?

Reply:
I moved the legacy changes into their own function.